src/test/java/jalau/at18/katas/game2048/team1/MainTest.java

```diff
@@ -0,0 +1,141 @@
+package jalau.at18.katas.game2048.team1;
+
+import org.junit.Test;
+import static org.junit.Assert.*;
+
+
+public class MainTest {
+    private final int[][] exampleMatrix = {{0, 0, 0, 0}, {0, 0, 0, 0}, {0, 0, 2, 0}, {0, 0, 0, 0}};
+    private final int[][] matrixGameOver = {{1, 2, 3, 4}, {5, 6, 7, 8}, {9, 1, 2, 3}, {4, 5, 9, 7}};
+    private final int[][] matrixGameNoOver = {{1, 2, 3, 4}, {5, 6, 7, 0}, {9, 1, 2, 4}, {4, 5, 9, 20}};
+
+    public static final int RANDOM_VALUE_ONE = 2;
+    public static final int RANDOM_VALUE2 = 4;
+    public static final int MATRIX_LENGTH = 4;
+    public static final int COUNT_ZEROS = 15;
+
+
+    @Test
+    public void mainCountZerosMatrix() {
+        Main main = new Main();
+        assertEquals(COUNT_ZEROS, main.countZerosMatrix(exampleMatrix));
+    }
+    @Test
+    public void mainGameOver() {
+        Main main = new Main();
+        assertEquals(true, main.gameOver(matrixGameOver));
+    }
+    @Test
+    public void mainGameNoOver() {
+        Main main = new Main();
+        assertEquals(false, main.gameOver(matrixGameNoOver));
+    }
+
+    @Test
+    public void mainRandomValue() {
+        Main main = new Main();
+        int num = main.randomValueGrid();
+        assertTrue(num == RANDOM_VALUE_ONE || num == RANDOM_VALUE2);
+    }
+
+    @Test
+    public void mainRandomPositionGrid() {
+        Main main = new Main();
+        int[] valuesPosition = main.randomPositionGrid();
+        assertTrue(valuesPosition[0] < MATRIX_LENGTH && valuesPosition[1] < MATRIX_LENGTH);
+    }
+
+    @Test
+    public void mainAddTile() {
+        Main main = new Main();
+        int countTiles1 = 0;
+        for (int row = 0; row < exampleMatrix.length; row++) {
+            for (int column = 0; column < exampleMatrix.length; column++) {
+                if (exampleMatrix[row][column] != 0) {
+                    countTiles1++;
+                }
+            }
+        }
+        int[][] matrixPlusTile = main.addTile(exampleMatrix);
+        int countTiles2 = 0;
+        for (int row = 0; row < matrixPlusTile.length; row++) {
+            for (int column = 0; column < matrixPlusTile.length; column++) {
+                if (matrixPlusTile[row][column] != 0) {
+                    countTiles2++;
+                }
+            }
+        }
+        assertEquals(countTiles1 + 1, countTiles2);
```

Comment:
Tests can be simple. If you already know that `exampleMatrix` has 1 tile, and after running `addTile()` you should have 2, then, the first `for` for counting initial number of tiles might not be needed, you can directly do:
```
assertEquals(2, countTiles)
```
Where `countTiles` is only the count of the tiles after `addTile` is called.

Reply:
We already correct this.